rpc/lib/client/http_client.go

```diff
@@ -83,35 +85,63 @@ func makeHTTPClient(remoteAddr string) (string, *http.Client) {
 
 //------------------------------------------------------------------------------------
 
+// jsonRPCBufferedRequest encapsulates a single buffered request, as well as its
+// anticipated response structure.
+type jsonRPCBufferedRequest struct {
+	request types.RPCRequest
+	result  interface{} // The result will be deserialized into this object.
+}
+
+// JSONRPCRequestBatch allows us to buffer multiple request/response structures
+// into a single batch request. Note that this batch acts like a FIFO queue, and
+// is thread-safe.
+type JSONRPCRequestBatch struct {
+	requests []*jsonRPCBufferedRequest
```

Comment:
the convention we're using in Tendermint - fields protected by the mutex should be declared after it. E.g.

```go
mtx      sync.Mutex
A typeA
B typeB

V typeC // fields, which are not protected by mutex, get separated by a new line
```